cmd/node/config/gasSchedules/gasScheduleV2.toml

```diff
@@ -0,0 +1,607 @@
+    DelegateExecution    = 160000
+    AsyncCallStep        = 200000
+    AsyncCallbackGasLock = 20000000
+    ExecuteReadOnly      = 160000
+    CreateContract       = 300000
+    GetReturnData        = 100
+    GetNumReturnData     = 100
+    GetReturnDataSize    = 100
+
+[EthAPICost]
+    UseGas              = 100
+    GetAddress          = 100000
+    GetExternalBalance  = 70000
+    GetBlockHash        = 100000
+    Call                = 160000
+    CallDataCopy        = 200
+    GetCallDataSize     = 100
+    CallCode            = 160000
+    CallDelegate        = 160000
+    CallStatic          = 160000
+    StorageStore        = 250000
+    StorageLoad         = 100000
+    GetCaller           = 100
+    GetCallValue        = 100
+    CodeCopy            = 1000
+    GetCodeSize         = 100
+    GetBlockCoinbase    = 100
+    Create              = 320000
+    GetBlockDifficulty  = 100
+    ExternalCodeCopy    = 3000
+    GetExternalCodeSize = 2500
+    GetGasLeft          = 100
+    GetBlockGasLimit    = 100000
+    GetTxGasPrice       = 1000
+    Log                 = 3750
+    GetBlockNumber      = 100000
+    GetTxOrigin         = 100000
+    Finish              = 1
+    Revert              = 1
+    GetReturnDataSize   = 200
+    ReturnDataCopy      = 500
+    SelfDestruct        = 5000000
+    GetBlockTimeStamp   = 100000
+
+[BigIntAPICost]
+    BigIntNew                = 2000
+    BigIntByteLength         = 2000
+    BigIntUnsignedByteLength = 2000
+    BigIntSignedByteLength   = 2000
+    BigIntGetBytes           = 2000
+    BigIntGetUnsignedBytes   = 2000
+    BigIntGetSignedBytes     = 2000
+    BigIntSetBytes           = 2000
+    BigIntSetUnsignedBytes   = 2000
+    BigIntSetSignedBytes     = 2000
+    BigIntIsInt64            = 2000
+    BigIntGetInt64           = 2000
+    BigIntSetInt64           = 2000
+    BigIntAdd                = 2000
+    BigIntSub                = 2000
+    BigIntMul                = 6000
+    BigIntTDiv               = 6000
+    BigIntTMod               = 6000
+    BigIntEDiv               = 6000
+    BigIntEMod               = 6000
+    BigIntAbs                = 2000
+    BigIntNeg                = 2000
+    BigIntSign               = 2000
+    BigIntCmp                = 2000
+    BigIntNot                = 2000
+    BigIntAnd                = 2000
+    BigIntOr                 = 2000
+    BigIntXor                = 2000
+    BigIntShr                = 2000
+    BigIntShl                = 2000
+    BigIntFinishUnsigned     = 1000
+    BigIntFinishSigned       = 1000
+    BigIntStorageLoadUnsigned   = 100000
+    BigIntStorageStoreUnsigned  = 250000
+    BigIntGetArgument           = 1000
+    BigIntGetUnsignedArgument   = 1000
+    BigIntGetSignedArgument     = 1000
+    BigIntGetCallValue          = 1000
+    BigIntGetExternalBalance    = 10000
+
+[CryptoAPICost]
+    SHA256          = 600
+    Keccak256       = 600
+    Ripemd160       = 600
+    VerifyBLS       = 1000
```

Comment:
Why is this so cheap? It is actually the hardest computation stuff we have in the protocol. Can we have a test that checks a couple of BLS signatures in the VM? Also, the rest of the crypto operations are calibrated?

Reply:
We have. Check the benchmarks and the corresponding C file. Actually this is executed rather fast.